Add list of disks endpoint (/api/diskList)

agafoner opened the pull request:
Created Vue application: /client
Added components

agafoner committed:
Add list of disks endpoint (/api/diskList)
Add list of files and dirs thought path (e.g. /api/DirList?path=C: )

StepanovAleksey commented on client/src/App.vue:
```diff
@@ -0,0 +1,60 @@
+<template>
+<div>
+  <TopBar></TopBar>
+  <div class="panelDiv">
```
названия классов идут через kebab-case, это не обязательно но это правило "хорошего тона"

StepanovAleksey commented on client/src/App.vue:
```diff
@@ -0,0 +1,60 @@
+  align-items: stretch;
+  align-content: center;
+}
+div {
```
аккуратнее. у тебя нет scoped атрибута т.е эти классы будут применяться ко всем элементам которые подходят под селекторы

StepanovAleksey commented on client/src/components/Panel.vue:
```diff
@@ -0,0 +1,32 @@
+<template>
+<div class="panel forDraft">
```
kebab-case

StepanovAleksey commented on client/src/components/Panel.vue:
```diff
@@ -0,0 +1,32 @@
+
+<style scoped>
+.panel{
+  height: 600px;
```
ай ай ай. Уходи от размеров в px. Всё на flex-ах. В твоём проекте можно вообще нигде размеры не указывать

StepanovAleksey commented on client/src/components/TopBar.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+<div class="topBar forDraft ">
+  <button style="width: 60px;height: 60px; float:left; margin-left: 10px;"> Settings</button>
```
почему не в стиле?

StepanovAleksey commented on client/src/components/TopBar.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+<div class="topBar forDraft ">
```
kebab-case

StepanovAleksey commented on client/src/components/Panel.vue:
```diff
@@ -0,0 +1,32 @@
+</div>
+</template>
+
+<script>
```
определись уже, используешь TS или нет. В первом компоненте используешь TS потом не используешь

StepanovAleksey commented on client/src/components/panel/FileElement.vue:
```diff
@@ -0,0 +1,27 @@
+
+<style scoped>
+.element{
+  width: calc(100% - 10px);
```
не очень понял почему так, нельзя было padding-ом обойтись?

StepanovAleksey reviewed (changes requested):
есть замечания

agafoner committed:
1 Переделал основные размеры приложения, постарался уйти от px (см App).
2 Стиль кнопок один - в App.
3 Во всех компонентах указал ts.
4 Привёл синтаксис к Options API в компонентах.

agafoner committed:
1 Переделал основные размеры приложения, постарался уйти от px (см App).
2 Стиль кнопок один - в App.
3 Во всех компонентах указал ts.
4 Привёл синтаксис к Options API в компонентах.

StepanovAleksey commented on client/src/components/panel/FileElement.vue:
```diff
@@ -0,0 +1,29 @@
+  name: "Element",
+  props: {
+    fileData: {
+      type: Array
```
если массив то fileDates

StepanovAleksey commented on client/src/components/panel/PanelBar.vue:
```diff
@@ -0,0 +1,23 @@
+</script>
+
+<style scoped>
+.panelBar{
```
kebabe-case
